File: app/src/main/java/team/triplog/presentation/activity/TripPlanActivity.java
package team.triplog.presentation.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.widget.Toolbar;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import team.triplog.R;
import team.triplog.presentation.adapter.TripPlanDayAdapter;

public class TripPlanActivity extends BaseActivity {
    private Toolbar toolbar;
    private ConstraintLayout buttonAccount;
    private ConstraintLayout buttonCheck;
    private ConstraintLayout buttonMember;
    private ConstraintLayout buttonPhoto;
    private TextView textTitle;
    private TextView textTripName;
    private TextView textTripPeriod;
    private RecyclerView recyclerTripPlan;
    private TripPlanDayAdapter planDayAdapter;
    private ArrayList<String> items = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_trip_plan);

        init();
        setData();
        setUi();
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        supportFinishAfterTransition();
    }

    @Override
    public void onEnterAnimationComplete() {
        super.onEnterAnimationComplete();
    }

    private void init() {
        toolbar = findViewById(R.id.toolbar);
        textTitle = findViewById(R.id.text_title);
        textTripName = findViewById(R.id.text_trip_name);
        textTripPeriod = findViewById(R.id.text_trip_period);
        buttonAccount = findViewById(R.id.button_account);
        buttonCheck = findViewById(R.id.button_check);
        buttonMember = findViewById(R.id.button_member);
        buttonPhoto = findViewById(R.id.button_photo);
        recyclerTripPlan = findViewById(R.id.recycler_trip_plan);

        LinearLayoutManager layoutManager = new LinearLayoutManager(this);

        planDayAdapter = new TripPlanDayAdapter(this, items);
        recyclerTripPlan.setLayoutManager(layoutManager);
        recyclerTripPlan.setAdapter(planDayAdapter);
        recyclerTripPlan.setNestedScrollingEnabled(false);

        setSupportActionBar(toolbar);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
            getSupportActionBar().setDisplayShowTitleEnabled(false);
        }

        buttonAccount.setOnClickListener(onClickListener);
        buttonCheck.setOnClickListener(onClickListener);
        buttonMember.setOnClickListener(onClickListener);
        buttonPhoto.setOnClickListener(onClickListener);
    }

    private void setData() {
        //더미데이터ㅎㅎ
        items.add("TEST_DAY1");
        items.add("TEST_DAY2");
        items.add("TEST_DAY3");
        items.add("TEST_DAY4");
        items.add("TEST_DAY5");

        planDayAdapter.setItems(items);
    }

    private void setUi() {
        // TODO : String 형식이 아닌 Trip 객체를 받아서 처리 필요.
        Intent intent = getIntent();
        textTitle.setText(intent.getStringExtra("EXTRA_TRIP_NAME"));
        textTripName.setText(intent.getStringExtra("EXTRA_TRIP_NAME"));
        textTripPeriod.setText(intent.getStringExtra("EXTRA_TRIP_PERIOD"));
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            supportFinishAfterTransition();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private View.OnClickListener onClickListener = view -> {
        switch (view.getId()) {
            case R.id.button_account:
                Intent intent = new Intent(this, TripAccountActivity.class);
                startActivity(intent);
                break;

            case R.id.button_check:
                intent = new Intent(this, TripCheckActivity.class);
                startActivity(intent);
                break;

            case R.id.button_member:
                intent = new Intent(this, TripMemberActivity.class);
                startActivity(intent);
                break;

            case R.id.button_photo:
                intent = new Intent(this, TripAlbumActivity.class);
                startActivity(intent);
                break;
        }
    };
}

File: app/src/main/java/team/triplog/presentation/activity/TripPlanActivity.kt
package team.triplog.presentation.activity

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.databinding.DataBindingUtil
import team.triplog.R
import team.triplog.databinding.ActivityTripPlanBinding
import team.triplog.presentation.adapter.TripPlanDayAdapter
import team.triplog.presentation.base.BaseActivity
import java.util.*

class TripPlanActivity : BaseActivity() {
    private lateinit var binding: ActivityTripPlanBinding

    private var planDayAdapter: TripPlanDayAdapter? = null
    private val items = ArrayList<String>()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_trip_plan)

        init()
        setData()
        setUi()
    }

    override fun onBackPressed() {
        super.onBackPressed()
        supportFinishAfterTransition()
    }

    override fun onEnterAnimationComplete() {
        super.onEnterAnimationComplete()
    }

    private fun init() {
        planDayAdapter = TripPlanDayAdapter(this, items)
        binding.recyclerTripPlan.apply {
            adapter = planDayAdapter
            isNestedScrollingEnabled = false
        }

        setSupportActionBar(binding.toolbar)

        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            setDisplayShowHomeEnabled(true)
            setDisplayShowTitleEnabled(false)
        }

        binding.buttonAccount.setOnClickListener(onClickListener)
        binding.buttonCheck.setOnClickListener(onClickListener)
        binding.buttonMember.setOnClickListener(onClickListener)
        binding.buttonPhoto.setOnClickListener(onClickListener)
    }

    private fun setData() {
        //더미데이터ㅎㅎ
        items.add("TEST_DAY1")
        items.add("TEST_DAY2")
        items.add("TEST_DAY3")
        items.add("TEST_DAY4")
        items.add("TEST_DAY5")
        planDayAdapter!!.setItems(items)
    }

    private fun setUi() {
        // TODO : String 형식이 아닌 Trip 객체를 받아서 처리 필요.
        binding.textTitle.text = intent.getStringExtra("EXTRA_TRIP_NAME")
        binding.textTripName.text = intent.getStringExtra("EXTRA_TRIP_NAME")
        binding.textTripPeriod.text = intent.getStringExtra("EXTRA_TRIP_PERIOD")
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            supportFinishAfterTransition()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    private val onClickListener = View.OnClickListener { view: View ->
        when (view.id) {
            R.id.button_account -> {
                Intent(this, TripAccountActivity::class.java).apply {
                    startActivity(this)
                }
            }
            R.id.button_check -> {
                Intent(this, TripCheckActivity::class.java).apply {
                    startActivity(this)
                }
            }
            R.id.button_member -> {
                Intent(this, TripMemberActivity::class.java).apply {
                    startActivity(this)
                }
            }
            R.id.button_photo -> {
                Intent(this, TripAlbumActivity::class.java).apply {
                    startActivity(this)
                }
            }
        }
    }
}